pkg/component/gardener/discoveryserver/discoveryserver.go

```diff
@@ -0,0 +1,190 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package discoveryserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	"github.com/gardener/gardener/pkg/component"
+	operatorclient "github.com/gardener/gardener/pkg/operator/client"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	secretsutils "github.com/gardener/gardener/pkg/utils/secrets"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+)
+
+const (
+	// ManagedResourceNameRuntime is the name of the ManagedResource for the runtime resources.
+	ManagedResourceNameRuntime = "gardener-discovery-server-runtime"
+	// ManagedResourceNameVirtual is the name of the ManagedResource for the virtual resources.
+	ManagedResourceNameVirtual = "gardener-discovery-server-virtual"
+
+	// DeploymentName is the name of the Gardener Discovery Server deployment.
+	DeploymentName = "gardener-discovery-server"
```

Comment:
same here?

Reply:
This is actually used here: https://github.com/gardener/gardener/blob/e1e2947f726b7ef1e2cdcd1f8f655fbc6b3935d0/pkg/operator/controller/garden/garden/components.go#L1299

Reply:
Technically, you should expose the ServiceName then, since the first arg of `NetworkPolicyLabel` takes a service name).